docs/references/sdk/frontend-only.mdx

```diff
@@ -1,7 +1,152 @@
 ---
-description: TODO
+description: A reference for implementing a frontend-only Clerk SDK
 ---
 
 # Frontend-only SDK
 
-TODO
+While [ClerkJS](/docs/references/javascript/overview) can be used in any browser context and framework, realistically users expect to consume its features through the conventions and syntax of their framework of choice. `@clerk/clerk-react` turns ClerkJS into React components, `@clerk/astro` into Astro components, and so on.
+
+In non-browser environments (e.g. iOS) you’ll need to re-implement the [Clerk class](/docs/references/javascript/clerk/clerk) in the SDK’s programming language, interacting with the [FAPI](/docs/references/sdk/terminology).
+
+## Expected features
+
+- User only needs to provide their [publishable key](/docs/references/sdk/terminology)
+- User only needs to adjust one or two files to add Clerk to their app (e.g. adding Clerk to the configuration file of that framework)
+- User can use [Clerk’s components](/docs/components/overview) in their choice of framework (e.g. in a React-based framework you import these components as React components)
+- Give users access to `client`, `session`, `user`, and `organization` properties through the framework’s choice of state management
+- User should be able to use ClerkJS options
+
+## Implementation
+
+While the implementation details will vary for each SDK, there are certain steps you'll have to go through in any case. Consider the steps below a rough guidance on what needs to be done, and also remember to follow the [conventions](/docs/references/sdk/conventions).
+
+> [!NOTE]
+> The code blocks below will be written in pseudo-code. If you're looking for real-world examples, have a look at these repositories: [`@clerk/clerk-react`](https://github.com/clerk/javascript/tree/main/packages/react), [`@clerk/astro`](https://github.com/clerk/javascript/tree/main/packages/astro)
+
+<Steps>
+  ### Create a Clerk instance
+
+  Create a Clerk instance that will only be invoked once (e.g. following the [singleton pattern](https://www.patterns.dev/vanilla/singleton-pattern/)). During its initialization you'll execute the following steps.
+
+  ```tsx {{ mark: [5], filename: 'create-clerk-instance.ts' }}
```

Comment:
In the future we may want to differentiate between `tsx` and `ts` in the UI, for example showing different icons for each. For this reason I think we should use `ts` here since it's explicitly a TS file, not TSX

```suggestion
  ```ts {{ mark: [5], filename: 'create-clerk-instance.ts' }}
```